posts/2022-10-19-runtime-multistage-programming.adoc

```diff
@@ -0,0 +1,326 @@
+# Custom JIT compilation with Runtime Multi-Stage Programming
+
+For a while now I've been fiddling with a library called https://github.com/markehammons/slinc[Slinc], a way to link C code in Scala. Since my original version, I've been using context values, macros, and other tools to generate code for writing and reading representations of Structs into and out of native memory.
+
+Originally, I was using Scala 3 macros to produce native bindings and helper code, but I recently discovered that Runtime Multi-Stage Programming produces much better code. My Slinc project therefore is a real-world example of the worth of Runtime Multi-Stage Programming
+
+## A review of Macros
+
+If you're not familiar with how macros work in Scala, the most simple way of writing them involves using quotes and splices.
+
+```scala
```

Comment:
I think code blocks in adoc are done differently https://docs.asciidoctor.org/asciidoc/latest/verbatim/source-blocks/.

Reply:
Done